ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh

```diff
@@ -24,14 +24,24 @@ if [[ "${CLUSTER_TYPE:-}" =~ ^aws-s?c2s$ ]]; then
   REGION=$(jq -r ".\"${LEASED_RESOURCE}\".source_region" "${CLUSTER_PROFILE_DIR}/shift_project_setting.json")
 fi
 
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
 # 1. get vpc id and public subnet
-VpcId=$(cat "${SHARED_DIR}/vpc_id")
+if [[ ! -f "${SHARED_DIR}/vpc_id" && ! -f "${SHARED_DIR}/public_subnet_ids" ]]; then
+  if [[ -e ${SHARED_DIR}/metadata.json ]]; then
```

Comment:
Once that, the vpc would fail to be deleted  on ipi-deprovision step, because the bastion become a dependency of the vpc.

Reply:
Correct, you'd need to add
```
    - ref: aws-deprovision-stacks
    - ref: aws-deprovision-s3buckets
```
before deprovision steps first

Reply:
for QE's existing provision chain where need a bastion, we always set customer vpc as a pre-condition, so that is not a problem for us. I meat, if you applied the step to a common IPI install, then need to do some adjustment for the ipi deprovsion chain.

Reply:
Yeah, I'm using additional steps in https://github.com/openshift/release/pull/42055, but I don't think its worth updating common ipi-deprovision chain for this.

I suppose just a warning in bastion ref should suffice?